lib/_http_server.js

```diff
@@ -1030,6 +1035,11 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
     if (req.upgrade)
       return 2;
   }
+  if(server.parseIncomingUrlWithWhatWg) {
+    req.originalUrl = req.url;
+    const newUrl = new URL(req.url, 'http://localhost');
```

Comment:
Wouldn't using the request host header be better than using localhost?

Reply:
True, but it depends on if we should expose the whole url instance

Reply:
+1 here, let's rebuild the original URL using the scheme and the host.

Reply:
+1 here, let's rebuild the original URL using the scheme and the host.